features/prog_test.go

```diff
@@ -92,3 +95,85 @@ func TestHaveProgTypeInvalid(t *testing.T) {
 		t.Fatalf("Expected os.ErrInvalid but was: %v", err)
 	}
 }
+
+func TestHaveProgHelper(t *testing.T) {
+	type testCase struct {
+		prog     ebpf.ProgramType
+		helper   asm.BuiltinFunc
+		expected error
+		version  string
+	}
+
+	// Referecing linux kernel commits to track the kernel version required to pass these test-cases
+	// These test-cases are derived from libbpf's selftests and helper/prog combinations that are
+	// probed for in cilium/cilium
+	// Still missing since those helpers are not available in the lib yet, are:
+	// - Kprobe, GetBranchSnapshot
+	// - SchedCLS, SkbSetTstamp
+
+	testCases := []testCase{
+		{ebpf.Kprobe, asm.FnMapLookupElem, nil, "3.19"},                     // @d0003ec
+		{ebpf.Kprobe, asm.FnKtimeGetCoarseNs, ebpf.ErrNotSupported, "5.16"}, // @5e0bc30
```

Comment:
Feel free to comment this one out until 5.17 testing is live in CI.

I've added a note to https://github.com/cilium/ebpf/pull/668 to make sure this gets uncommented before merging.

Reply:
Commented two test cases out, which require CI kernel changes.